src/main/kotlin/com/linchpino/ai/service/impl/ResumeService.kt

```diff
@@ -0,0 +1,72 @@
+package com.linchpino.ai.service.impl
+
+import com.linchpino.ai.model.Resume
+import com.linchpino.ai.repository.ResumeRepository
+import com.linchpino.core.exception.ErrorCode
+import com.linchpino.core.exception.LinchpinException
+import org.apache.pdfbox.Loader
+import org.apache.pdfbox.text.PDFTextStripper
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Service
+import java.io.File
+import java.io.IOException
+import java.util.*
+
+@Service
+class ResumeService(private val resumeRepository: ResumeRepository) {
+    private val logger: Logger = LoggerFactory.getLogger(ResumeService::class.java)
+
+    @Value("\${spring.ai.roadmap-attempt-limit}")
+    private val roadmapAttemptLimit = 0
+
+    fun create(file: File?): Resume {
+        requireNotNull(file) { "File cannot be null" }
+        val lines = readLines(file)
+        val email = Resume.findEmail(lines)
```

Comment:
Resume.findEmail delagates to Contact.findEmail, since both are public and static you should directly call Contact.findEmail

Reply:
You’re right; I thought that repeatedly calling resume to locate the email might improve readability.